test/shopify.test.js

```diff
@@ -276,5 +276,57 @@ describe('Shopify', () => {
       return shopify.request(url, 'GET')
         .then(res => expect(res).to.deep.equal({}));
     });
+
+    it('times out requests (socket open)', () => {
+      const shopify = new Shopify({
+        shopName,
+        apiKey,
+        password,
+        timeout: 500
+      });
+
+      const url = assign({ path: '/test' }, shopify.baseUrl);
+
+      nock(`https://${shopName}.myshopify.com`)
+        .get('/test')
+        .socketDelay(1000)
+        .reply(200, {});
+
+      return shopify.request(url, 'GET')
+        .then(() => {
+          throw new Error('Request did not fail');
+        })
+        .catch((err) => {
+          expect(err.message).to.equal(
+            'Socket timed out on request to undefined'
```

Comment:
Ok, I guess the `req._headers.host` property does not exists, see https://github.com/floatdrop/timed-out/blob/ead538ea1b5a11c979d8632276b05a36859dbd76/index.js#L19, still curious though.